components/log-viewer-webui/client/src/api/socket/MongoCollectionSocket.ts

```diff
@@ -0,0 +1,50 @@
+import {
+    ClientToServerEvents,
+    ServerToClientEvents,
+} from "@common/index.js";
+import {Socket} from "socket.io-client";
+
+import {MongoCursorSocket} from "./MongoCursorSocket.js";
+import {getSharedSocket} from "./SocketSingleton.js";
+
+
+/**
+ * Socket connection to a MongoDB collection residing on a server. Class provides methods to
+ * query the collection.
+ */
+class MongoCollectionSocket {
+    collectionName: string;
+
+    private socket: Socket<ServerToClientEvents, ClientToServerEvents>;
```

Comment:
let's use the JS ES6 private member syntax instead of the TS one
```suggestion
    #socket: Socket<ServerToClientEvents, ClientToServerEvents>;
```